### PR TITLE
Issue 860

### LINKED ISSUE #118: Problems with the 3D model of the wings
The 3D model of the wings need some rework:

1) Glass reflection colliding with vent:

fgfs_screen_001

2) The wings are too thin, compare the photograph below with our plane:

3) Our wings are missing the drooped tips ("inverted winglets"), see:

### LINKED ISSUE #430: Better UV mapping
The fuselage could have some improved UV mapping in several places:

* * *

1) Tail needs better UV mapping, particularly between the rudder and the tail itself. This is very evident with the OH-CTL livery:

fgfs-screen-001

fgfs-screen-002

In the first picture above, note how the yellow stripe does not match between the rudder and the tail. In the second pitcure, note how there is a number missing in the yellow stripe (it was supposed to be written 15000 instead of 1500) and how the blue airplane logo has a mismatching left wing tip (the blue bit)

* * *

2) The back of the fuselage doesn't look good on the right side of the plane. This is quite evident with the livery N35799. See:

fgfs-screen-001

The other side isn't perfect but is much more acceptable:

fgfs-screen-002

* * *

3) The nose is also quite problematic, see:

fgfs-screen-001

### LINKED ISSUE #860: small 3D issues with the external fuselage
These are a couple of minor 3D issues that we currently have with our external fuselage:

- While testing the tutorials, the camera angle for removing the tie-down in the tail of the plane on the preflight tutorial shows a possible problem with the 3D model, see:

fgfs-screen-001

On the left side of the purple circle we can see through the fuselage! (click on the image to open it in higher resolution to see what I am talking about)

- problematic UV mapping near the aircraft nose in the left side, colors look really strange on all liveries:

fg-0

fg-1

- the rudder shape is a tiny bit wrong, it is missing a half-circle hole, see:

rudder

some other references:

### LINKED ISSUE #965: Some internal objects needs smoothing
Some internal objects needs smoothing:

### LINKED ISSUE #966: Back tip of the fuselage has strange reflection
Back tip of the fuselage has strange reflection, it is brighter than the fuselage from nearly all angles.

### PR DESCRIPTION
**Pull request by wlbragg**
Fixes #860
Fixes #430 
Fixes #966
Fixes #965 
Fixes #118

This is not ready to merge yet!

### CODE REVIEW

**Comment by gilbertohasnofb** [comment]
@wlbragg thanks again for tackling this huge issue! _Really_ appreciate it.

> So, please look at this in the sim and give me your impression, both interior and exterior and tell me what you think. Is this back to where it is supposed to be?

So concerning smoothing, most of it has been fixed but I spotted some potential problems, see pics below. The livery issues seem to be solved (wings and interior are now correct).

> the small square on the fuselage [...] In other words it was apparently by design but we can get rid of it if you want.

yes please, remove that! I don't see it in photographs and it looks really ugly.

So here are the pics of the (potential) issues I found after testing, including some new ones. Some of these might now be issues at all but intentional things in which case you can ignore them.

fg-1
fg-2
fg-3
fg-4
fg-5
fg-6
fg-7
fg-8
fg-9
fg-10
fg-11
fg-12
fg-13
fg-0

**Comment by wlbragg** [comment]
@gilbertohasnofb @onox 
This Blend file is so buggard it isn't funny, there are numerous issues. The gap showing the sky through the fuselage near the exhaust is because the fuselage has been offset from where it is supposed to be which means the gear and everything else in not lined up anymore.
I don't know when, how or why any of this happened but this model is really screwed up.
@gilbertohasnofb some of the objects you think may not be smoothed are. I think they just have a lot of hard edges that make it appear to be otherwise.
Issues with any mesh that are not a part of this Blend such as fire extinguisher, headphones, buckle on the tie-down and the corners of the yoke are still outstanding.
The last picture "what are these?? Were they always there?" no, those are the pontoon or float struts and they were fixed a long time ago. Another thing that got buggard up somehow.

Long story short I think I got it back together.
Please check it out again and we'll go from there. Remember, fire extinguisher, headphones, buckle on the tie-down and the corners of the yoke are still outstanding.

**Comment by onox** [comment]
I've moved "Fixes #860" from the title to the description.

**Comment by gilbertohasnofb** [comment]
Oh, and this is also still a problem:



Is this part of the main model?

**Comment by wlbragg** [comment]
The seat frame supports are finished but I don't see them either, see I have proof.
seat
I'll look into why not.

>>The last picture "what are these?? Were they always there?" no, those are the pontoon or float struts and they were fixed a long time ago. Another thing that got buggard up somehow.

>I see. I just retested it and these are still a problem on my side.

I don't see them in my model. Can you verify that they are the pontoon floats. To test, watch that area while switching from wheels to floats or pontoons.

The fuel caps were not uv mapped, they were assigned to a texture we don't even use. So I UV mapped them to the wings. You should see them change to match the livery except I forgot to add their object to the livery section.

>Also, there is another issue (#430) related to the UV-mapping of the main model: tail not fitting and messy UV map on nose. Do you think these could also be taken care with this same PR?

Yes, actually some of that (tail especially) should already be cleaned up.

**Comment by wlbragg** [comment]
The fuselage part of the tail is as good as I am going to get it. I haven't looked at the uv mapping of the rudder portion yet but I will. Notice the tie-down fastener now (second pic), those objects were never added to the bumpmap effect, fuel caps either. Thats all corrected now, including the addition to the liveries so the cap will change color with some of the different liveries.
fgfs-screen-013
fgfs-screen-014

**Comment by wlbragg** [comment]
@gilbertohasnofb 
By the way, the seat frame problem was I only did 2 of the 3 seat frame objects. I missed a seat.

**Comment by gilbertohasnofb** [comment]
@wlbragg Ok, so I did one more check and I can confirm these are fixed:

- base of pilot seat now has the top polygons
- much better UV mapping of the tail!
- much better UV mapping of the nose!
- fuel tank caps are red again in some liveries (but not in others, see below)

To be done if possible:

- make the fuel tank caps red independent of texture or livery, just like it was before. This way we don't need to change every single livery and every single xcf file manually. Would that be possible? You mentioned that before we simply had them as untextured full red objects, and that was looking quite all right for such a tiny detail. 
- take a look if those modifications concerning the reflection of the tail of the fuselage (https://github.com/c172p-team/c172p-detailed/issues/966) have been pushed as I still see the bug on my side
- improve a tiny bit more the nose UV-mapping if possible, see pics below (if you think that's not possible it is fine with me, but there is one bit in particular which still look quite messy)
- check the grey bit by the bottom of the doors on the interior, perhaps map it to the same textures that surround it on the door and interior walls. This grey bit is not the pontoon by the way, this is present in all variants.

fg-5
fg-6
fg-3
fg-4
fg-7
fg-0
fg-1
fg-2

**Comment by onox** [comment]
I checked the exterior of some liveries:

- PT-IAO: the orange stripes on the flaps don't align with the stripes on the wings. The orange strip on the rudder (right side) doesn't align with the orange stripe on the tail. The lower part of the rudder (both sides) doesn't align with the tail.

- 4X-CHV: the blue stripe on the rudder (right side) doesn't align with the tail. The fuel caps are black instead of red.

- OH-CTL: OK

I see with multiple liveries that the lower right side of the rudder has alignment issues.

**Comment by onox** [comment]
The trim wheel is very dark, I can't see its surface.

**Comment by onox** [comment]
The air vents are not smooth.

**Comment by gilbertohasnofb** [comment]
@wlbragg About what @onox wrote:

> 4X-CHV: the blue stripe on the rudder (right side) doesn't align with the tail.

I think that the following would be a good way of solving this: do not move the UV-mapping but rather try to stretch it a bit downwards, using the livery 4X-CHV as reference. So using the top points as fixed, stretch it at the bottom downwards until it matches and then I am sure all other liveries will match again.

fg-1
  
> The air vents, fire extinguisher, headset are not smooth I think.

@onox see https://github.com/c172p-team/c172p-detailed/issues/965, @wlbragg said he will take a look at those separately in a different PR in the future, if I understood it correctly.

**Comment by wlbragg** [comment]
@gilbertohasnofb 
Please check the rudder correction I just pushed.

>he will take a look at those separately in a different PR in the future, if I understood it correctly.

Yes, you understood it correctly.

**Comment by gilbertohasnofb** [comment]
@wlbragg the rudder is looking good to me, thanks for the fix!

**Comment by wlbragg** [comment]
>PT-IAO: the orange stripes on the flaps don't align with the stripes on the wings. 

I think this may be the only issue I haven't addressed yet. All other have been addressed an my action or inaction explained. So please review and correct me if I am wrong.

**Comment by gilbertohasnofb** [comment]
@wlbragg ok, we are nearly there!

The good stuff:

- amazing fuselage in the back, bravo!!!
- flaps mapping seems fine now
- fire extinguisher is smooth
- yokes are now smooth and look much better
- fuel tank caps are red in all liveries 
- amazing new design of wings around the caps, great work

Things which could be taken care before merging this:

- trim wheel is still untextured, but we actually have a texture for it:
trim

- though the UV-mapping of the floor around the doors is muuuuch better, there is still a small issue with the right hand side:
fg-2
fg-3

- there is a tiny strange polygon in the fuselage by the bottom of the baggage door, could you please check what's that?:
fg-4
fg-5

We are very nearly there...... 👍 👍 👍

**Comment by wlbragg** [comment]
OK, will work on the mesh issues.

>trim wheel is still untextured, but we actually have a texture for it:

It is using that texture but it is just a black rectangle and thus were not able to see the detail in the mesh as it is all the same shade. So I was wondering is if we could make a better texture, one with some definition to it?

**Comment by wlbragg** [comment]
>there is a tiny strange polygon in the fuselage by the bottom of the baggage door,

This is truly an anomaly. I couldn't find any reason for it. But I fixed it by remapping the mesh faces involved even though they already appeared to be correct.

**Comment by wlbragg** [comment]
All this work got me ambitious so I decided to fixed the front glass so the vents look correct. I hope you all like it as it was something we've been contemplating for quite awhile.

**Comment by wlbragg** [comment]
fgfs-screen-001
fgfs-screen-002
fgfs-screen-003
fgfs-screen-004
fgfs-screen-005
fgfs-screen-006

**Comment by gilbertohasnofb** [comment]
> All this work got me ambitious [...]

Music to my ears 😄 

> [...] so I decided to fixed the front glass so the vents look correct. I hope you all like it as it was something we've been contemplating for quite awhile.

This is looking great, so much better!

As for the trim tab:

> It is using that texture but it is just a black rectangle and thus were not able to see the detail in the mesh as it is all the same shade. So I was wondering is if we could make a better texture, one with some definition to it?

Just map it to that texture and I will work on it tonight, together with the new mechanical texture for the tail. I will be out during the whole day but I will definitely take a look at this in the evening.

**Comment by gilbertohasnofb** [comment]
Just pushed a new trim texture, it's very close to the position of the previous one. I only applied it to the 4X-CHV interior texture, so please do not make it livery dependent. I will work on the mechanical tip of the fuselage later today.

**Comment by gilbertohasnofb** [comment]
@wlbragg super job, the wings are looking so much better!

But I did find some new problems:

fg-1
fg-2
fg-4
fg-7
fg-9
fg-0

I will now create the texture for the back tip of the fuselage, I will let you know once it is pushed.

**Comment by wlbragg** [comment]
Thanks, will look at these issues.

**Comment by wlbragg** [comment]
The rudder light is a good catch. Nothing to do with my recent work. There was an ALS light object and a rudder object (rudder_0) not assigned to the rotation animation. I guess they never were. It's like this in fgdata, I fixed it.

The strange polygon in the pilot side glass, I'm not seeing it and looking at the glass layers they are not touching or any closer than any of the rest of it. Please check it again after you pull the next revision and see if it is still there.

**Comment by wlbragg** [comment]
>Wings have strange mesh on the bottom side close to flaps.

This one is not so easy on my end. It is caused by the texture not extending the shaded gray area all the way to the top of the fuselage or frame. I don't know if it needs livery work too? Remapping would not work because it would make the door outline incorrect so we need the texture worked on.
white

**Comment by wlbragg** [comment]
>Bottom part of this texture
...
strange polygon or untextured bit?

Another good catch, these are old issues. There was an extra face down there that needed deleted and the existing one had the bottom portion indeed untextured. The strange polygon was caused by the floor needing to be extended ever so slightly.
Will push corrections with the wingtips.

**Comment by wlbragg** [comment]
>Bottom part of this texture
[snip]
strange polygon or untextured bit?

Another good catch, these are old issues. There was an extra face down there that needed deleted and the existing one had the bottom portion indeed untextured. The strange polygon was caused by the floor needing to be extended ever so slightly.
Will push corrections with the wingtips.

**Comment by wlbragg** [comment]
All done. I sure hope you like this version of the wings and wing tips better than the original because I would hate to have done this in vain. This was about at the limits of my ability with Blender. I would have like to designed it a little better, but working with an existing mesh made it really difficult to do. Plus the wingtips design I am trying to replicate is an extremely challenging shape. All in all though we have wingtips that are not all screwed up even if they aren't the exact shape we were shooting for.
I checked all the liveries and they are all OK. The original UV mapping of the wings was a couple pay grades beyond mine so I had to cheat a bit to get the tips UV mapped. I think it looks OK though.
Please let me know what you think.

**Comment by wlbragg** [comment]
fgfs-screen-004

**Comment by gilbertohasnofb** [comment]
@wlbragg thanks for the fixes, we are so nearly there :smile: You fixed almost everything from my last report, I believe there is only the wing mesh issue to be discussed plus some other minor things I spotted.

> This one is not so easy on my end. It is caused by the texture not extending the shaded gray area all the way to the top of the fuselage or frame.

Are you sure this is caused by the textures? I am 99% sure this looks like a 3D problem to me. Basically, this is what I think one of the sides of the wing should look without the flap and the next picture is what I believe is the problem:

wing1

wing2

wing3

**Comment by gilbertohasnofb** [comment]
fg-0
fg-1
fg-3
fg-4

**Comment by gilbertohasnofb** [comment]
As for the wing tips, they look really good! The mesh on top of it by the lamp looks excellent and your fix for the floor of the cockpit seems to have done the trick with that polygon. I think these above are the truly last issues before merging this. Thanks so much for your patience with all these trillion reports, I really appreciate it.

**Comment by gilbertohasnofb** [comment]
And just confirming, this here is a feature and not a mesh bug, right? It's just so thin and pointy that it made me think it could be a mesh problem.

fg

**Comment by dany93** [comment]
Impressive....
@gilbertohasnofb observations + @wlbragg : you are also a Blender expert!!

**Comment by wlbragg** [comment]
>Are you sure this is caused by the textures? I am 99% sure this looks like a 3D problem to me.

We're both right. It is exactly because of what your saying, but...

The fix I was asking for only fixed part of it. I didn't realize the back edge of the wing was also exposed and that face is 90 deg to the aft direction same as the tail face thus it really lights up when the sun is behind us. My fix wouldn't have corrected that anyway.

wing1

Problems:
I can't straiten that angle your highlighting to 90deg like it would have to be or the edge of the wing where it enters the fuselage pulls away from the fuselage.

If I straighten the angle from what it is now to as far as I can go, it covers the area I recommended for the texture modification making at least that part not necessary. But it would still leave some of the back edge of the wing visible.

So I tried to at least change the angle enough to cover the exposed white on the chassis frame which would eliminate any need for a texture modification and I guess it didn't like the extreme angles or the type of angles at play and it crashed the simulator as soon as i tried to change the view.
So I tried a second time and at least it didn't crash this time, but it makes me extremely nervous that it did the first time when I moved that piece of mesh. Plus moving that wing edge mesh as far as I can still doesn't gain us enough.

wing

Another thing, the wing thickening covered up most of the issue requiring the texture change, but not the exposed back face of the wing. But, you can only see it when on the ground by changing your view angle to be from ground level or below ground level. Thus you can really only see this now when flying in chase view with the sun behind you.

Summary:
The risk of moving that particular mesh and causing other CPU's to crash scares me, especially considering the gain we see from that.
Moving that wing edge like I did also technically changed the underside wing surface which undoubtedly changed the texturing under the wing. I never looked at that consequence.
I have no good answer as what we should do.

**Comment by wlbragg** [comment]
>And just confirming, this here is a feature and not a mesh bug, right?

Yes, that is the transponder antenna. Keep an eye out for a  good reference image and I will make a new one. Your right, his one does look more like a stray vert than an antenna..

**Comment by wlbragg** [comment]
>lamp is slightly misaligned

I can't believe you saw that. Yes, it was way worse so I did adjust it some and so from my perspective I convinced myself the mechanic that installed the lamp couldn't get it to fit perfect. OK, I'll fix the mechanics screwup.:)

**Comment by wlbragg** [comment]
I resized the windows so that is finished.

The anomaly around the top right of the baggage area going up to the window, I took a good look at that and tried to slightly adjust the area. This type of modification is beyond my ability or patience. There is so much of it I don't want to go down that road. The bottom of the baggage area is worse in my opinion and there is nothing I can do about that.

I also noticed something I never noticed before, the interior chassis is supposed to be sized to meet the fuselage precisely, well it is considerably flawed. In places you can see the z-fighting between chassis and fuselage. In other places you can see small gaps, like looking up from down low at the front top of the door and door windows. I don't know the procedure for making these part fit better, I guess that would be done from the start and maybe in this case it wasn't. This is the best case argument fro starting over from scratch and having a "professional modeler" make us a new model in the future.

**Comment by wlbragg** [comment]
>Perhaps this should be a single face

I thought the same thing. The difficult part is redesigning that area. Its hard to see and it is starting with a really weird shaped mesh. I need to look at that option a little closer. I would like to see a really good reference picture to the area. I did a cursory search for one and couldn't find any.

**Comment by wlbragg** [comment]
<UV map does not match behind flaps

Yep, my mistake, I didn't know there would be livery texture at that spot.

I'll take a look at the glass and I remember the ELT being iffy with some of its design so I'll take la look at is it also.

**Comment by gilbertohasnofb** [comment]
And let me know about the antenna. As for the livery wing shadow, I am fixing it right now, should be ready to push in some 30 minutes.

**Comment by wlbragg** [comment]
OK, I'll add the antenna to this, everything else should be done as of my last commit.

Good catch on the back glass, it had a vert that had been substantially moved to the wrong position.

**Comment by onox** [comment]
@wlbragg Did you also move the tie-down themselves? If you did, the FDM and some nasal also need to be updated.

**Comment by wlbragg** [comment]
No, I was able to move only the connectors on the wings. I was even able to better line them up with the tie-down loops so it should be good to go. I was thinking about making a new eye bolt for the tail like in that picture you provided but it would need to be repositioned farther back on the tail and then we would have to adjust the positions of the straps in the code and I didn't know if it was worth having to do that.

**Comment by gilbertohasnofb** [comment]
@wlbragg I pushed the modifications to the textures, so please do a `git pull` before continuing. I fixed all liveries _except_ for the older ones that were not created by me, as they do not have layers and are less trivial to fix (and to be honest, I think they are much poorer than the HD ones). So what is left to be fixed except the antenna? Are the tie-downs fine?

By the way, if you still have patience, I noticed that the UV mapping below and above the wings do not match that perfectly:

fgfs-screen-001
fgfs-screen-002

**Comment by wlbragg** [comment]
>so please do a git pull before continuing

Will do.

>So what is left to be fixed except the antenna? 

Nothing, with the exception of the bottom and top wing texture alignment.

Yes, the tie-downs or fine.

I'll get this finish up a little later tonight. It's supper time! :)

**Comment by gilbertohasnofb** [comment]
> Nothing, with the exception of the bottom and top wing texture alignment.

I did not see any modifications to the antenna though, did you push them? It still looks just like a pointy triangle and not like that reference photo.

> I'll get this finish up a little later tonight. It's supper time! :)

No rush! Enjoy your food 😄 Here it's already past 1am, so I might only be able to check anything else tomorrow. But if you make these modifications I will merge this first thing in the morning after testing it 😄

**Comment by wlbragg** [comment]
>I did not see any modifications to the antenna though, did you push them? It still looks just like a pointy triangle and not like that reference photo.

No, I still have that to do.

**Comment by wlbragg** [comment]
All done! New strut steps and connections to fuselage.

**Comment by wlbragg** [comment]
Oops, wrong button.

**Comment by wlbragg** [comment]
Antenna and everything else is finished, as far as I know.

**Comment by wlbragg** [comment]
fgfs-screen-006

fgfs-screen-007

**Comment by gilbertohasnofb** [comment]
@wlbragg Bravo!! :clap: Stunning job, really. Thank you so much for all the patience and for fixing every single thing that has been pointed out, we have a much stronger 3D model right now!

I will open/update issues with things that were not done on this PR (e.g. rudder cables, damaged wing, etc.). And whenever you feel like tackling the cockpit position just give me a shout, I will be happy to help.

**Comment by gilbertohasnofb** [comment]
@wlbragg @onox Can I delete this branch?

**Comment by wlbragg** [comment]
>Can I delete this branch?

OK by me.
Thank you for all the encouragement and the time it takes to finding this stuff and explain it!

**Comment by gilbertohasnofb** [comment]
> f you don't need a branch anymore, you can delete it. You don't need my permission for it wink

:+1:
